controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -231,162 +234,158 @@ def test_properties(self):
 
-    def test_get_host_by_identifiers_raise_multiplehostsfounderror(self):
-        host_munch_ret_1 = Munch({'id': 'host_id_1', 'name': 'test_host_1',
-                                  'iscsi_name': 'iqn.test.1'})
-        host_munch_ret_2 = Munch({'id': 'host_id_2', 'name': 'test_host_2',
-                                  'iscsi_name': 'iqn.test.3'})
-        host_munch_ret_3 = Munch({'id': 'host_id_3', 'name': 'test_host_3',
-                                  'WWPN': ['Test_wwn']})
-        ret1 = [host_munch_ret_1, host_munch_ret_2]
-        ret2 = Munch(as_single_element=host_munch_ret_2)
-        ret3 = Munch(as_single_element=host_munch_ret_3)
-        self.svc.client.svcinfo.lshost.side_effect = [ret1, ret2, ret3]
+    @patch("controller.array_action.svc_cli_result_reader.SVCListResultsReader.__iter__")
+    def test_get_host_by_identifiers_raise_multiplehostsfounderror(self, result_reader_iter):
+        host_1 = self._get_host_as_dictonary('host_id_1', 'test_host_1', ['iqn.test.1'], [])
+        host_2 = self._get_host_as_dictonary('host_id_2', 'test_host_2', ['iqn.test.3'], [])
+        host_3 = self._get_host_as_dictonary('host_id_3', 'test_host_3', [], ['Test_wwn'])
+        hosts = [host_1, host_2, host_3]
+        self.svc.client.svcinfo.lshost = Mock()
+        self.svc.client.svcinfo.lshost.return_value = self._get_hosts_list_result(hosts)
+        self.svc.client.send_raw_command = Mock()
+        self.svc.client.send_raw_command.return_value = EMPTY_BYTES, EMPTY_BYTES
+        result_reader_iter.return_value = self._get_detailed_hosts_list_result(hosts)
         with self.assertRaises(array_errors.MultipleHostsFoundError):
             self.svc.get_host_by_host_identifiers(Initiators('iqn.test.3', ['Test_wwn']))
 
-    def test_get_host_by_identifiers_return_iscsi_host(self):
-        host_munch_ret_1 = Munch({'id': 'host_id_1', 'name': 'test_host_1',
-                                  'WWPN': ['abc1']})
-        host_munch_ret_2 = Munch({'id': 'host_id_2', 'name': 'test_host_3',
-                                  'iscsi_name': ['iqn.test.2'],
-                                  'WWPN': ['abc3']})
-        host_munch_ret_3 = Munch({'id': 'host_id_3', 'name': 'test_host_3',
-                                  'WWPN': ['abc3'],
-                                  'iscsi_name': 'iqn.test.3'})
-        ret1 = [host_munch_ret_1, host_munch_ret_2]
-        ret2 = Munch(as_single_element=host_munch_ret_2)
-        ret3 = Munch(as_single_element=host_munch_ret_3)
-        self.svc.client.svcinfo.lshost.side_effect = [ret1, ret2, ret3]
+    @patch("controller.array_action.svc_cli_result_reader.SVCListResultsReader.__iter__")
+    def test_get_host_by_identifiers_return_iscsi_host(self, result_reader_iter):
+        host_1 = self._get_host_as_dictonary('host_id_1', 'test_host_1', [], ['abc1'])
+        host_2 = self._get_host_as_dictonary('host_id_2', 'test_host_2', ['iqn.test.2'], ['abc3'])
```

Comment:
host2 is 'abc3'?

Reply:
abc3 is wwn.
This was original test value. I ddidn't change it